### PR TITLE
ref(performance-metrics): Not fire metrics request with transaction.duration [INGEST-759]

### PR DESCRIPTION
**Pull request by priscilawebdev**
- [x] Add new prop `onSwitch` to the `MetricsSwitch` component
- [x] Clear URL query when switching from transaction x metrics
- [x] Move `eventView` out of the state

### CODE REVIEW

**Review by matejminar — COMMENTED** [review]
Looks like `transaction.duration` is still sent.

https://user-images.githubusercontent.com/9060071/145409866-33ba0193-6895-4944-ada9-e2b2db9b8a99.mp4

File: static/app/views/performance/content.tsx [hunk 1]
```diff
@@ -92,7 +78,11 @@ function PerformanceContent({selection, location, demoMode}: Props) {
     }
   }, [selection.datetime]);
 
-  const {eventView, error} = state;
+  const {error} = state;
+
+  const eventView = generatePerformanceEventView(location, organization, projects, {
```
Comment (priscilawebdev) [inline static/app/views/performance/content.tsx:83]: this does not need to be in the state

File: tests/js/spec/views/performance/metricsSwitch.spec.tsx [hunk 2]
```diff
@@ -8,11 +8,13 @@ import {
...
 } from 'sentry/views/performance/metricsSwitch';
 
+const handleSwitch = jest.fn();
+
 function TestComponent() {
   const {isMetricsData} = useMetricsSwitch();
   return (
     <Fragment>
-      <MetricsSwitch />
+      <MetricsSwitch onSwitch={handleSwitch} />
```
Comment (matejminar) [inline tests/js/spec/views/performance/metricsSwitch.spec.tsx:17]: Can we add an assertion in `it('toggles between transactions and metrics')` test to make sure that handleSwitch mock gets called?

**Review by matejminar — APPROVED** [review]
Can confirm that the problem we were having with wrong queries is fixed.
Will leave the final approval for the performance team.

**Pull request merged**